vyper/parser/memory_allocator.py

```diff
@@ -0,0 +1,24 @@
+from typing import (
+    Tuple,
+)
+
+from vyper.utils import (
+    MemoryPositions,
+)
+
+
+class MemoryAllocator():
+    next_mem: int
+
+    def __init__(self, start_position: int = MemoryPositions.RESERVED_MEMORY):
+        self.next_mem = start_position
+
+    # Get the next unused memory location
```

Comment:
Actually, one other thing.  I've noticed that we're often using comments above method and function definitions when it's more customary to use docstrings.  Can we switch to docstrings or is there a particular reason it's being done this away?

Reply:
@davesque main reason is that was the convention on repo when I joined, also I think it reads well, especially because there is a difference between comments and docs. Comments augment the code or 'narrative' of the code were reading the code alone isn't good enough, docblocks is good for understanding how to call a function. The fact that docblocks are required to be fat by convention, supports the point. A good balance between docblocks and comments is what I prefer over a codebase with too much docblocks. But that's just my completely biased and super objective opinion.